SignalMessaging/environment/migrations/OWSDatabaseMigrationRunner.m

```diff
@@ -56,52 +56,42 @@ - (void)assumeAllExistingMigrationsRun
 
 - (void)runAllOutstandingWithCompletion:(OWSDatabaseMigrationCompletion)completion
 {
-    [self runMigrations:self.allMigrations completion:completion];
+    [self runMigrations:[self.allMigrations mutableCopy] completion:completion];
 }
 
-- (void)runMigrations:(NSArray<OWSDatabaseMigration *> *)migrations
+// Run migrations serially to:
+//
+// * Ensure predictable ordering.
+// * Prevent them from interfering with each other (e.g. deadlock).
```

Comment:
Fixed a serious bug here and generally made the migration process more predictable and legible in the logs.

Reply:
Good call. Thanks for serializing these.